### PR TITLE
HEEDLS-276 Diagnostic assessment launch back end

### PR DESCRIPTION
**Pull request by kwunkiuma**
Create query for retrieving data for content viewer and add tests.

Ran and passed all unit tests.

### CODE REVIEW

File: DigitalLearningSolutions.Data.Tests/Services/DiagnosticAssessmentServiceTests.cs [hunk 1]
```diff
@@ -238,22 +235,137 @@ int progressId
...
-
-                // Then
-                sectionInCourseContent.SectionScore.Should().Be(scoresReturnedFromOldStoredProcedure.SecScore);
-                sectionInCourseContent.MaxSectionScore.Should().Be(scoresReturnedFromOldStoredProcedure.SecOutOf);
-                sectionInCourseContent.DiagnosticAttempts.Should().Be(scoresReturnedFromOldStoredProcedure.DiagAttempts);
-            }
+            // Given
+            var scoresReturnedFromOldStoredProcedure = diagnosticAssessmentTestHelper
+                .ScoresFromOldStoredProcedure(progressId, sectionId)
+                .FirstOrDefault();
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Data.Tests/Services/DiagnosticAssessmentServiceTests.cs:241]: Why is `FirstOrDefault` necessary here? 🙂

**Review by FridaTveit — COMMENTED** [review]
One comment, otherwise looks good 🙂

File: DigitalLearningSolutions.Data.Tests/Services/DiagnosticAssessmentServiceTests.cs (hunk 1 above)
Comment (kwunkiuma) [reply DigitalLearningSolutions.Data.Tests/Services/DiagnosticAssessmentServiceTests.cs:241]: It didn't occur to be to change this after the comment for the post learning PR, I'll move this logic to the helper function.

**Review by FridaTveit — APPROVED** [review]
Looks good 🙂

**Pull request merged**